code/modules/clothing/suits/marine_armor.dm

```diff
@@ -490,196 +431,84 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
-		overlays += I
-	else armor_overlays["lamp"] = null
-	if(user) user.update_inv_wear_suit()
-
-/obj/item/clothing/suit/storage/faction/pickup(mob/user)
-	if(flags_faction_armor & ARMOR_LAMP_ON && src.loc != user)
-		user.SetLuminosity(brightness_on)
-		SetLuminosity(0)
-	..()
-
-/obj/item/clothing/suit/storage/faction/dropped(mob/user)
-	if(flags_faction_armor & ARMOR_LAMP_ON && src.loc != user)
-		user.SetLuminosity(-brightness_on)
-		SetLuminosity(brightness_on)
-		toggle_armor_light() //turn the light off
-	..()
+//===========================//UPP\\================================
 
-/obj/item/clothing/suit/storage/faction/Dispose()
-	if(ismob(src.loc))
-		src.loc.SetLuminosity(-brightness_on)
-	else
-		SetLuminosity(0)
-	. = ..()
-
-/obj/item/clothing/suit/storage/faction/attack_self(mob/user)
-	if(!isturf(user.loc))
-		to_chat(user, "<span class='warning'>You cannot turn the light on while in [user.loc].</span>")
-		return
-
-	if(flashlight_cooldown > world.time)
-		return
-
-	if(!ishuman(user)) return
-	var/mob/living/carbon/human/H = user
-	if(H.wear_suit != src) return
-
-	toggle_armor_light(user)
-	return 1
 
-/obj/item/clothing/suit/storage/faction/item_action_slot_check(mob/user, slot)
-	if(!ishuman(user)) return FALSE
-	if(slot != WEAR_JACKET) return FALSE
-	return TRUE //only give action button when armor is worn.
-
-/obj/item/clothing/suit/storage/faction/proc/toggle_armor_light(mob/user)
-	flashlight_cooldown = world.time + 20 //2 seconds cooldown every time the light is toggled
-	if(flags_faction_armor & ARMOR_LAMP_ON) //Turn it off.
-		if(user) user.SetLuminosity(-brightness_on)
-		else SetLuminosity(0)
-	else //Turn it on.
-		if(user) user.SetLuminosity(brightness_on)
-		else SetLuminosity(brightness_on)
-
-	flags_faction_armor ^= ARMOR_LAMP_ON
-
-	playsound(src,'sound/machines/click.ogg', 15, 1)
-	update_icon(user)
-
-	update_action_button_icons()
-
-
-
-
-/obj/item/clothing/suit/storage/faction/UPP
+/obj/item/clothing/suit/storage/marine/UPP
 	name = "\improper UM5 personal armor"
-	desc = "Standard body armor of the UPP military, the UM5 (Union Medium MK5) is a medium body armor, roughly on par with the venerable M3 pattern body armor in service with the USCM. Unlike the M3, however, the plate has a heavier neckplate, but unfortunately restricts movement slightly more. This has earned many UA members to refer to UPP soldiers as 'tin men'."
+	desc = "Standard body armor of the UPP military, the UM5 (Union Medium Mk 5) is a medium body armor, roughly on par with the M3 pattern body armor."
 	icon_state = "upp_armor"
 	slowdown = SLOWDOWN_ARMOR_MEDIUM
-	flags_armor_protection = UPPER_TORSO|LOWER_TORSO
+	flags_marine_armor = ARMOR_LAMP_OVERLAY
 	armor = list(melee = 60, bullet = 60, laser = 50, energy = 60, bomb = 40, bio = 10, rad = 10)
-	uniform_restricted = list(/obj/item/clothing/under/marine/veteran/UPP)
 
-/obj/item/clothing/suit/storage/faction/UPP/commando
+/obj/item/clothing/suit/storage/marine/UPP/commando
```

Comment:
Perhaps if you want to unify all clothes into a tag `military` might be better than `marine`.

Reply:
This would be greatly appreciated.